AstSemantics.md

```diff
@@ -241,14 +234,48 @@ Accesses to global variables specify the index as an integer literal.
 The specification will add atomicity annotations in the future. Currently
 all global accesses can be considered "non-atomic".
 
+## Control flow structures
+
+WebAssembly offers basic structured control flow. All control flow structures
+are statements.
+
+  * `block`: a fixed-length sequence of statements
+  * `if`: if statement
+  * `do_while`: do while statement, basically a loop with a conditional branch
+    (back to the top of the loop)
+  * `forever`: infinite loop statement (like `while (1)`), basically an
+    unconditional branch (back to the top of the loop)
+  * `continue`: continue to start of nested loop
+  * `break`: break to end from nested loop or block
+  * `return`: return zero or more values from this function
+  * `switch`: switch statement with fallthrough
+
+Break and continue statements can only target blocks or loops in which they are
+nested. This guarantees that all resulting control flow graphs are well-structured.
```

Comment:
`switch` doesn't allow Duff's device, even though it supports fallthrough, because the AST structure prevents nesting a loop which straddles `switch` statement blocks.